arkouda/strings.py

```diff
@@ -828,17 +828,114 @@ def save(self, prefix_path : str, dataset : str='strings_array',
         self.bytes.save(prefix_path=prefix_path, 
                                     dataset='{}/values'.format(dataset), mode=mode)
 
-    def register(self, user_defined_name : str) -> Strings:
+    def register(self, user_defined_name: str) -> Strings:
+        """
+        Register this Strings object with a user defined name in the arkouda server
+        so it can be attached to later using Strings.attach()
+        This is an in-place operation, registering a Strings object more than once will
+        update the name in the registry and remove the previously registered name.
+        A name can only be registered to one object at a time.
+
+        Parameters
+        ----------
+        user_defined_name : str
+            user defined name which the Strings object is to be registered under
+
+        Returns
+        -------
+        Strings
+            The same Strings object which is now registered with the arkouda server and has an updated name.
+            This is an in-place modification, the original is returned to support a fluid programming style.
+            Please note you cannot register two different objects with the same name.
+
+        Raises
+        ------
+        TypeError
+            Raised if pda is neither a pdarray nor a str or if
```

Comment:
I think we can remove the pda & pdarray line here... (both here and in the corresponding pdarray class method since we are only taking in a string argument, so the user_defined_name can stay).

Reply:
Okay, I can go ahead and remove it for `strings`. Should I also go ahead and remove it from `pdarrayclass` or open a separate issue for changing those `pdarrayclass` docstrings?

Reply:
My opinion is that it's ok to fix the documention in pdarrayclass as well.

If @mhmerrill agrees it's ok to modify the code for the `typechecked` annotations he mentioned above, then I say this ticket should be updated to add the type-checking in both `Strings` and `pdarrayclass` along with the documentation updates.

Reply:
yes, I think that is good.